orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -179,6 +193,30 @@ class OperationsController {
     }
   }
 
+  private void recordPipelineFailure(Map pipeline, String errorMessage) {
+    Exception pipelineError = null
+    try {
+      pipeline = parseAndValidatePipeline(pipeline)
```

Comment:
What's the purpose of calling `parseAndValidatePipeline` here if we already know that the pipeline is failing?  (I'd imagine we might not even expect the pipeline to be valid if we caught an exception while preparing it in `echo`.)

Reply:
yeah, i was going back and forth on that... My thoughts were: if it does `parse` then it will be nicer from the UI perspective as the pipeline will be rendered "fully/correctly" (this especially applied to MPT pipelines)

If it fails, on the other hand, I figured we can capture both errors for the user to fix.

Do you think it's reasonable?

Reply:
Thanks for the explanation; yeah, that sounds reasonable to me

Reply:
👍 will add a comment to the code as well